src/coreclr/src/jit/lowerxarch.cpp

```diff
@@ -1925,6 +1925,18 @@ void Lowering::ContainCheckStoreLoc(GenTreeLclVarCommon* storeLoc)
     assert(storeLoc->OperIsLocalStore());
     GenTree* op1 = storeLoc->gtGetOp1();
 
+    if (op1->OperIs(GT_BITCAST))
+    {
+        // If we know that the source of the bitcast will be in a register, then we can make
```

Comment:
How does it work when bitcast generate a real move from one register to another:
https://github.com/dotnet/runtime/blob/f99b4fe46497185db3b2049913f2af0c839fda00/src/coreclr/src/jit/codegenxarch.cpp#L1764-L1766
for cases like `int a = foo(); b = doo(long a);`.
Can the bitcast, in this case, be marked as contained?

Reply:
This isn't a case where we would use `BITCAST` - it is only for places where we "reinterpret" bits as a different type of the same size.
But by making the `BITCAST` nodes contained where possible, it reduces the cases where we needlessly manifest a value in a register when it could be loaded or stored directly.

Reply:
> This isn't a case where we would use BITCAST - it is only for places where we "reinterpret" bits as a different type of the same size.

That code from codegenxarch.cpp `genCodeForTreeNode: case GT_BITCAST` generates a real move and I saw that this code path is not unreachable.
or the new case that you have added in this PR in `codegenarmarch.cpp:genCodeForTreeNode(GenTree* treeNode):case GT_BITCAST`  it also generates a real move from one register to another.
I do not understand how we avoid marking such bitcasts as contained.

Reply:
Yes, we will generate a real move. Consider the case where we have a `RETURN` that returns a `Vector2` in an integer register. In that case the `BITCAST` itself would not be contained (that only happens for things that might be stores). If the source is a `SIMD` node it won't be contained either, and will be generated into a float register, and the `BITCAST` will generate a move to the return register. This is also what happens for the case of a lclVar source that is RegOptional and gets a register. It will be living in a float register, and the `BITCAST` will move the value to the return register.